File: server/feed13/src/feed/services/inbound.clj
(ns feed.services.inbound
  (:require [feed.daos.inbound :as i]
  	    [feed.daos.cache :as c]))

(defn fetch
  "fetch the inbound news feed items for a participant"
  [id]
  (c/get id i/fetch))

(defn create
  "create an inbound news feed item for a participant"
  [from to occurred subject story]
  (i/create from to occurred subject story))

File: client/flutter/lib/models/inbound.dart
import './outbound.dart';
import './participant.dart';

class InboundModel extends OutboundModel {
  final ParticipantModel from;
  InboundModel({
    required this.from,
    required occurred,
    required subject,
    required story,
  }) : super(occurred: occurred, subject: subject, story: story);
}

class InboundEnvelope {
  List<InboundModel> build(Map<String, dynamic> payload) {
    final List<InboundModel> retVal = [];
    payload['data']['inbound'].forEach((item) {
      retVal.add(InboundModel(
        from: ParticipantModel(
	  id: '',
	  name: item['from']['name'],
	),
        occurred: item['occurred'],
        subject: item['subject'],
        story: item['story'],
      ));
    });
    return retVal;
  }
}